intern/src/lib.rs

```diff
@@ -130,74 +116,32 @@ macro_rules! intern {
 }
 
 lazy_static::lazy_static! {
-    static ref INTERNED: (ArcSwap<HashSet<ArenaStr>>, Mutex<(HashSet<ArenaStr>, Bump)>)
+    static ref INTERNED: (ArcSwap<HashSet<InternedArenaStr>>, Mutex<(HashSet<InternedArenaStr>, Bump)>)
         = (ArcSwap::new(Arc::new(HashSet::new())), Mutex::new((HashSet::new(), Bump::new())));
 }
 
 pub fn preloaded<T: InternString>(value: &str) -> Option<T> {
-    let set = INTERNED.0.load();
-    unsafe { Some(T::to_interned(*set.get(value)?)) }
+    InternedArenaStr::preloaded(value).map(|s| unsafe { T::to_interned(s) })
```

Comment:
Similar to above, it would be great to document the use of `unsafe` and why this particular usage is safe. We should make sure we document all usages of unsafe thoroughly.